lib/ace/snippets.js

```diff
@@ -914,8 +915,6 @@ var TabstopManager = function(editor) {
         var ranges = this.ranges;
         tabstops.forEach(function(ts, index) {
             var dest = this.$openTabstops[index] || ts;
```

Comment:
When you assign `[]` to $openTabstops, why don't you assign `tabstops` to it instead? Then you don't have to slowly assign elements from `tabstops` to it later on. Note that you would also have to push all the elements from `tabstops` to `arg`.

Btw, can you give `arg` a different name? I have no idea what it is. Similar for `i`.